stl/inc/ranges

```diff
@@ -6821,6 +6822,496 @@ namespace ranges {
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                auto _Modified_iterator = _Lhs;
+                _Modified_iterator -= _Rhs;
+
+                return _Modified_iterator;
+            }
+
+            _NODISCARD_FRIEND constexpr difference_type
+                operator-(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                    noexcept(_Zip_get_smallest_iterator_sentinel_distance<difference_type>(
+                        _STD declval<_Iterator_tuple_type>(), _STD declval<_Iterator_tuple_type>()))) // strengthened
+                requires(sized_sentinel_for<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>,
+                    iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+                return _Zip_get_smallest_iterator_sentinel_distance<difference_type>(_Lhs._Current, _Rhs._Current);
+            }
+
+            _NODISCARD_FRIEND constexpr auto iter_move(const _Iterator& _Itr) noexcept(
+                (noexcept(_RANGES iter_move(_STD declval<const iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>()))
+                    && ...)
+                && (is_nothrow_move_constructible_v<
+                        range_rvalue_reference_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...)) {
+                return _Tuple_transform(_RANGES iter_move, _Itr._Current);
+            }
+
+            friend constexpr void iter_swap(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                (noexcept(_RANGES iter_swap(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>(),
+                     _STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                    && ...)) requires(indirectly_swappable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+                const auto _Swap_every_pair_lambda = [&_Lhs, &_Rhs ]<size_t... _Indices>(index_sequence<_Indices...>) {
+                    ((_RANGES iter_swap(_STD get<_Indices>(_Lhs._Current), _STD get<_Indices>(_Rhs._Current))), ...);
+                };
+
+                _Swap_every_pair_lambda(index_sequence_for<_ViewTypes...>{});
+            }
+        };
+
+        template <bool _IsConst>
+        class _Sentinel {
+        private:
+            using _Sentinel_tuple_type = tuple<sentinel_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+            /* [[no_unique_address]] */ _Sentinel_tuple_type _End;
+
+            friend zip_view;
+
+            constexpr explicit _Sentinel(_Sentinel_tuple_type _End) noexcept(
+                is_nothrow_copy_constructible_v<_Sentinel_tuple_type>) // strengthened
+                : _End(_End) {}
+
+            template <bool _IteratorConst>
+            using _Iterator_sentinel_difference_type =
+                common_type_t<range_difference_t<_Maybe_const<_IteratorConst, _ViewTypes>>...>;
+
+            template <bool _IteratorConst>
+            using _Iterator_type = typename zip_view<_ViewTypes...>::template _Iterator<_IteratorConst>;
+

```

Comment:
These types don't seem necessary to me (especially `_Iterator_type<_C>` could just be written `_Iterator<_C>` below)